ironic_prometheus_exporter/parsers/ipmi.py

```diff
@@ -0,0 +1,306 @@
+import re
+
+# NOTE (iurygregory): most of the sensor readings come in the ipmi format
+# each type of sensor consider a different range of values that aren't integers
+# (eg: 0h, 2eh), 0h will be published as 0 and the other values as 1, this way
+# we will be able to create prometheus alerts.
+# Documentation: https://www.intel.com/content/www/us/en/servers/ipmi/
+# ipmi-second-gen-interface-spec-v2-rev1-1.html
+
+
+def add_prometheus_type(name, metric_type):
+    return '# TYPE %s %s' % (name, metric_type)
+
+
+def metric_names(payload, prefix, sufix, extract_unit=False,
+                 special_label=None):
+    metric_dic = {}
+    for entry in payload:
+        if special_label == 'fan':
+            e = re.sub(r'[\d].*$', '', entry.lower())
+            e = re.sub(r'[\(\)]', '', e).split()
+            label = '_'.join(e)
+        else:
+            e = re.sub(r"[\d]+", "", entry).lower().split()
+            label = '_'.join(e[:-1]).replace('-', '_')
+
+        if extract_unit:
+            sensor_read = payload[entry]['Sensor Reading'].split()
+            if len(sensor_read) > 1:
+                sufix = '_' + sensor_read[-1].lower()
+
+        if special_label == 'memory':
+            if 'mem' not in label and 'memory' not in label:
+                label = 'memory_' + label
+
+        metric_name = prefix + label + sufix
+        if metric_name in metric_dic:
+            metric_dic[metric_name].append(entry)
+        else:
+            metric_dic[metric_name] = [entry]
+    return metric_dic
+
+
+def extract_labels(entries, payload, node_name):
+    default_label = 'node_name="%s"' % node_name
+    if len(entries) == 1:
+        return {entries[0]: '{%s}' % default_label}
+    entries_labels = {}
+    for entry in entries:
+        try:
+            sensor = payload[entry]['Sensor ID'].split()
+            sensor_id = str(int(re.sub(r'[\(\)]', '', sensor[-1]), 0))
+            metric_label = [default_label,
+                            'sensor="%s"' % (sensor[0] + sensor_id)]
+            entries_labels[entry] = '{%s}' % ','.join(metric_label)
+        except Exception as e:
+            print(e)
```

Comment:
I think there was already a comment from @dtantsur about logging, but I don't remember if it was this line or another.

Reply:
Yeah, going to update =)